File: src/0253-Meeting-Rooms-II/0253.cpp
class Solution 
{
public:
    int minMeetingRooms(vector<Interval>& intervals) 
    {
        map<int, int> m;
        for (auto& it : intervals) 
        {
            m[it[0]]++, m[it[1]]--;
        }

        int res = 0, n = 0;
        for (auto& it : m) 
        {
            n += it.second;
            res = max(res, n); 
        }
        return res;
    }
}

File: src/0253-Meeting-Rooms-II/0253.py
class Solution:
    def minMeetingRooms(self, intervals):
        if not intervals:
            return 0
        tmp = sorted(x for i, j in intervals for x in [[i, 1], [j, -1]])
        res, n = 0, 0
        for _, v in tmp:
            n += v
            res = max(res, n)
        return res